### DISCUSSION TITLE
Operation of 2/3/4 way valve

### DISCUSSION THREAD

**Discussion by niekveldhuizen** [raomin/ESPAltherma#189] [Q&A]
Hi,

I am using the "def/ALTHERMA(LT_DA_04-08KW).h" file. Does some one know which defenition is nessecary for detecting which way the valve is opperating. For space heating or DHW heating?

I am using {0x10,0,217,1,-1,"Operation Mode"}, and that gives me "Heating/Cooling/Fan Only" but when it is set to heating it does not say of it heating the DHW tank or the underfloor heating.

**Comment by DieterTHeck** [comment]
Hi,

Why not try and load all valve entries and monitor?
For me 3-way valve switches between DHW and heating.

Or is that not what you meant?

![grafik](https://user-images.githubusercontent.com/58054042/202328839-0f3e31bc-d322-40f4-b041-b0f243235974.png)

**Reply by niekveldhuizen** [reply]
Yes that is what I mean. 

My next step was to try al valve values yesterday, but have not found time to do that. Busy soldering temperture sensors for my floor heating distribution valves 

Will try the valve values because i see 2/3/4 way versions en check what the do.

**Comment by raomin** [comment]
You should find in the definition something like this:
```
1 | 0x60,12,306,1 | 4way Valve(Space/DHW) |  
18 | 0x60,12,307,1 | 2way valve(On:Heat_Off:Cool) |  
1 | 0x60,12,307,1 | 3way Valve(Cool/Heat)
```
https://github.com/raomin/ESPAltherma/wiki/Information-about-Values
